src/include/sof/audio/kpb.h

```diff
@@ -59,6 +59,7 @@
 
 enum kpb_state {
 	KPB_STATE_BUFFERING = 0,
+	KPB_STATE_DRAINING,
 	KPB_STATE_DRAINING_ON_DEMAND,
```

Comment:
@keyonjie, I think it is now understandable and according to cAVS naming convention:
`KPB_STATE_BUFFERING ` -> kpb is buffering incoming data into history buffer
`KPB_STATE_DRAINING` -> kpb is draining data out of internal history buffer to host (no buffering or any other copy is performed in this state)
`KPB_STATE_DRAINING_ON_DEMAND` -> kpb copies incoming, real time data directly to host. No buffering.

Reply:
@mrajwa I assume that this change replaces #1435? I must agree with @keyonjie. `KPB_STATE_DRAINING_ON_DEMAND` sounds like some kind of different type of draining. `KPB_STATE_COPYING` makes much more sense. Also I'm not sure how are we addressing the problem with potential loss of data between start and end of draining. @lbetlej Was it agreed to fix it in the future PRs?

Reply:
> `KPB_STATE_DRAINING_ON_DEMAND` sounds like some kind of different type of draining

According to cAVS documentation it is another version of draining. Guys I just wanted to code according to that document so we are consistent across platforms. However since you feel like this is not very clear we can change `KPB_STATE_DRAINING_ON_DEMAND ` to `KPB_STATE_HOST_COPY` would it be OK for you?

PS: I assume all other KPB state are fine for you and we only disscuss the `KPB_STATE_DRAINING_ON_DEMAND ` ?

Reply:
@mrajwa That would be OK.